### PR TITLE
Prefer Undici over Axios

### PR DESCRIPTION
**Pull request by AverageHelper**
Since [Node 18 includes the `fetch` API](https://nodejs.org/en/blog/announcements/v18-release-announce#fetch-experimental) (and that support is [no longer experimental](https://stackoverflow.com/a/75221620)), we can drop a dependency when we upgrade. Since Node's `fetch` implementation is `undici`, I think it would be prudent to use that one for now. Once we're on Node 18, we can remove `undici` (or [move it into devDependencies for type information](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/60924#issuecomment-1504635244)) for a smaller bundle.

Axios is a fine package, as far as I can tell, and very popular, even since Node 18. The `fetch` API is also popular (see [`node-fetch`](https://github.com/node-fetch/node-fetch) and the browser-native [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/fetch) API). I consider both to be industry standard REST client implementations. Since Node has added the `fetch` API natively, thus presenting a chance to drop an external dependency, I'd prefer to go that route.

## Notable Changes
- Replaced `axios` with `undici`, in preparation for Node 18.
- Used `superstruct` to create type guards at API boundaries.
- Log HTTP status messages along with non-200 status codes.

### CODE REVIEW

**Review by Plyb — APPROVED** [review]

**Review by Plyb — APPROVED** [review]

File: src/helpers/HttpStatusCode.ts [hunk 1]
```diff
@@ -0,0 +1,409 @@
+/**
```
Comment (JstnMcBrd) [inline src/helpers/HttpStatusCode.ts:1]: This is a lot of information we may never use. Is there not a dependency that includes type codes so we don't have to define them ourselves?

Comment (AverageHelper) [reply src/helpers/HttpStatusCode.ts:1]: There are several, but none as complete as I wanted. The top two results on NPM didn't include HTTP 530, which I have observed from `findRoom` myself.

This file is largely copypasta from other projects of mine. At some point, I plan to either deploy it as an NPM package, or contribute to an already-popular package.

Comment (AverageHelper) [reply src/helpers/HttpStatusCode.ts:1]: It'd be real nice of Node packaged constants for at least the standard HTTP codes and their names, but alas, they do not.

Comment (AverageHelper) [reply src/helpers/HttpStatusCode.ts:1]: Also, I like my hover text to be descriptive, rather than needing to google a ref for HTTP status codes, lol

**Pull request merged**